tests/e2e/example.test.e2e.js

```diff
@@ -29,6 +29,18 @@ test('Test Register Form', async ({page}, workerInfo) => {
   save_visual(page);
 });
 
+test('Test Login Form OpenID navigation', async ({page}, workerInfo) => {
```

Comment:
I know that e2e tests are good.

However, I didn't see how this test really tests .... because e2e test isn't really used in Gitea.


Ideally, the test could be done in "integration tests", or if it is too difficult, I think it's better to remove this e2e test because it doesn't really help if anything breaks.

Reply:
It's good to have some e2e tests, but currently I'm not sure whether they would even fail the CI as there is some kind of "flaky" mechanism going on.

Reply:
imho such extensive tests would be totally out of scope for this PR

Reply:
But what does this e2e test do? I do not think it makes sense.

Reply:
maybe it's just basic but it tests that button is visible and is clickable and navigates to expected page.

Reply:
> maybe it's just basic but it tests that button is visible and is clickable and navigates to expected page.

Why not test it in backend "integration tests"? A correctly constructed link could always redirect to expected page.

I don't see how e2e tests really help Gitea's situation at the moment ...... few people ever wrote useful e2e tests for long time. If there is no determination to correctly and widely use the e2e testing system, my preference is to keep the testing system simple. Just my personal opinion.

Reply:
At the moment I agree with that but if more and more htmlx is used this could be not so easy anymore and e2e tests will need to be done

Reply:
> but it tests that button is visible and is clickable and navigates to expected page.

And if you agree with this, I think my comment above for "test the buttons shouldn't be shown if the OIDC is disabled" also makes sense, it is not "totally out of scope for this PR", but directly related to this PR's change.

Reply:
We can not test e2e tests with different app.ini options

Reply:
That's why I think "integration tests" are better .........

Reply:
they have the same issue, we have hard coded app.ini for tests and we can test only these scenarios...

Reply:
Actually we can mock (almost) everything in backend code, so no issue in integration tests. Many tests change the "setting" variables to test various cases.